### PR TITLE
feat(providers): add page and remove overview page

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes

Closes https://github.com/Qiskit/qiskit.org/issues/2963

## Implementation details

- Removed Overview page and added Providers page by:
  - reusing the overview page structure for the new providers page
  - renamed all variables, paths, and components from overview -> providers
  - removed unused code 
  - removed `TheQuickStart` section 
  - kept all assets and content (for now), until content is finalized
  - added some content shown in the Figma design
  
### Note:
I'm making the PR into the [feat/providers](https://github.com/Qiskit/qiskit.org/tree/feat/providers) feature branch, given that there'll be more work to be done here. 

In addition, we'll need to coordinate the launch of this page, with the updates to the UI Shell navigation menu, which will be handled in our web-components repo.

### CODE REVIEW

**Review by eddybrando — APPROVED** [review]
LGTM! Thank you very much @techtolentino 🙂

File: components/providers/ContentAccordion.vue [hunk 1]
```diff
@@ -27,7 +27,7 @@
 <script lang="ts">
 import Vue from 'vue'
 import { Component, Prop } from 'vue-property-decorator'
-import { AccordionLayoutProps } from '~/components/overview/AccordionLayout.vue'
+import { AccordionLayoutProps } from './AccordionLayout.vue'
```
Comment (y4izus) [inline components/providers/ContentAccordion.vue:30]:
```suggestion
import { AccordionLayoutProps } from '~/components/providers/AccordionLayout.vue'
```

**Review by y4izus — CHANGES_REQUESTED** [review]
Just a small change and we can merge, thank you Randy!

File: components/providers/ContentAccordion.vue (hunk 1 above)
Comment (techtolentino) [reply components/providers/ContentAccordion.vue]: Updated in this commit https://github.com/Qiskit/qiskit.org/pull/2964/commits/85154b31a7e84597e6e5e2c5d6454f5c9ae45e31

**Review by y4izus — APPROVED** [review]

**Pull request merged**